src/test/java/io/github/bobocodebreskul/context/registry/AnnotatedBeanDefinitionReaderTest.java

```diff
@@ -0,0 +1,142 @@
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.atLeastOnce;
+import static org.mockito.Mockito.atMostOnce;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+import io.github.bobocodebreskul.context.annotations.Autowired;
+import io.github.bobocodebreskul.context.annotations.BringComponent;
+import io.github.bobocodebreskul.context.config.AnnotatedGenericBeanDefinition;
+import io.github.bobocodebreskul.context.config.BeanDefinition;
+import io.github.bobocodebreskul.context.registry.AnnotatedBeanDefinitionReader;
+import io.github.bobocodebreskul.context.registry.BeanDefinitionRegistry;
+import java.util.Collections;
+import java.util.List;
+import org.apache.commons.lang3.StringUtils;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.MethodOrderer;
+import org.junit.jupiter.api.Order;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestMethodOrder;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mockito;
+
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
+public class AnnotatedBeanDefinitionReaderTest {
+
+  private BeanDefinitionRegistry registry;
+  private AnnotatedBeanDefinitionReader annotatedBeanDefinitionReader;
+
+  @BeforeEach
+  void init() {
+    registry = Mockito.spy(BeanDefinitionRegistry.class);
+    annotatedBeanDefinitionReader = new AnnotatedBeanDefinitionReader(registry);
+  }
+
+  @Test
+  @DisplayName("Test simple bean definition is created and passed to registry")
+  @Order(1)
+  void givenBeanClass_WhenRegisterBean_ThenBeanDefinitionCreatedAndPassedToRegistry() {
+    //given
+    var beanClass = MyComponent.class;
+    ArgumentCaptor<String> nameCaptor = ArgumentCaptor.forClass(String.class);
+    ArgumentCaptor<BeanDefinition> definitionCaptor = ArgumentCaptor.forClass(BeanDefinition.class);
+
+    //when
+    annotatedBeanDefinitionReader.registerBean(beanClass);
+
+    //then
+    verify(registry, atLeastOnce()).isBeanNameInUse(anyString());
+    verify(registry, atMostOnce()).registerBeanDefinition(anyString(), any());
+    verify(registry).registerBeanDefinition(nameCaptor.capture(), definitionCaptor.capture());
+    assertThat(nameCaptor.getValue()).isEqualTo(
+        StringUtils.uncapitalize(beanClass.getSimpleName()));
+    BeanDefinition actualBeanDefinition = definitionCaptor.getValue();
+    assertThat(actualBeanDefinition).isInstanceOf(AnnotatedGenericBeanDefinition.class);
+    assertThat(actualBeanDefinition.getBeanClass()).isEqualTo(beanClass);
+    assertThat(actualBeanDefinition.getDependsOn()).isEmpty();
+    assertThat(actualBeanDefinition.isAutowireCandidate()).isFalse();
+    assertThat(actualBeanDefinition.isPrimary()).isFalse();
+    assertThat(actualBeanDefinition.isSingleton()).isTrue();
```

Comment:
I advice to use here `.as(...)` method, it will give more detailed message what is wrong if test will fail. For example if `assertThat(actualBeanDefinition.isAutowireCandidate()).isFalse()` will fail, you will see that expected something to be `false` but was `true` and we have here two `.isFalse` checks, but with `.as(...)` you will see your additional message and it will be clearer to understand what has been failed.

In result it will look like this 
`assertThat(actualBeanDefinition.isAutowireCandidate()).as("Assert autowire candidate is false").isFalse()`